### ISSUE TITLE
Node 12 breaks URL parser

### ISSUE THREAD

**Issue by Stumblinbear** [deepstreamIO/deepstream.io-client-js#486]
Upgrading to NodeJS v12 causes the following connection issue. Server has no output.

```--- You can catch all deepstream errors by subscribing to the error event ---
............/node_modules/deepstream.io-client-js/dist/lib/client.js:204
    throw new Error(errorMsg);
    ^

Error: connectionError: Error: unexpected server response (400) (C)
    at Client._$onError (............/node_modules/deepstream.io-client-js/dist/lib/client.js:204:11)
    at Timeout._onTimeout (............/node_modules/deepstream.io-client-js/dist/lib/message/connection.js:319:19)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)```

**Comment by yasserf** [comment]
Thanks! We are looking into it

**Comment by yasserf** [comment]
This should be fixed in 2.3.1 👍
